test/gtest/ucs/test_memtype_cache.cc

```diff
@@ -278,3 +286,109 @@ UCS_TEST_P(test_memtype_cache, diff_mem_types_diff_bufs_keep_mem) {
 
 INSTANTIATE_TEST_CASE_P(mem_type, test_memtype_cache,
                         ::testing::ValuesIn(mem_buffer::supported_mem_types()));
+
+class test_memtype_cache_deferred_create : public test_memtype_cache {
+protected:
+    virtual void init() {
+        /* do nothing */
+    }
+
+    void test_unknown_region_found(const mem_buffer &b) const {
+        if (b.mem_type() == UCS_MEMORY_TYPE_HOST) {
+            return;
+        }
+
+        test_ptr_found(b.ptr(), b.size(), UCS_MEMORY_TYPE_LAST);
+    }
+};
+
+UCS_TEST_P(test_memtype_cache_deferred_create, allocate_and_update) {
+    const size_t size = 1000000;
+    void *ptr;
+
+    {
+        mem_buffer b(size, GetParam());
+
+        test_memtype_cache::init();
+
+        test_unknown_region_found(b);
+        test_region_not_found(b);
+
+        memtype_cache_update(b.ptr(), b.size(), GetParam());
```

Comment:
in other tests we use b.mem_type() instead of GetParam()

Reply:
fixed